plugins/inputs/solr/solr.go

```diff
@@ -0,0 +1,433 @@
+package solr
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const mbeansPath = "/admin/mbeans?stats=true&wt=json&cat=CORE&cat=QUERYHANDLER&cat=UPDATEHANDLER&cat=CACHE"
+const adminCoresPath = "/solr/admin/cores?action=STATUS&wt=json"
+
+type node struct {
+	Host string `json:"host"`
+}
+
+const sampleConfig = `
+  ## specify a list of one or more Solr servers
+  servers = ["http://localhost:8983"]
+
+  ## specify a list of one or more Solr cores (default - all)
+  # cores = ["main"]
+`
+
+// Solr is a plugin to read stats from one or many Solr servers
+type Solr struct {
+	Local       bool
+	Servers     []string
+	HTTPTimeout internal.Duration
+	Cores       []string
+	client      *http.Client
+}
+
+// AdminCores is an exported type that
+// contains a response with information about Solr cores.
+type AdminCoresStatus struct {
+	Status map[string]struct {
+		Index struct {
+			SizeInBytes int64 `json:"sizeInBytes"`
+			NumDocs     int   `json:"numDocs"`
```

Comment:
I would use int64 thoughout, all int fields are store das int64 internally and this can prevent differences with 32-bit systems.  It is not required though if the value cannot be larger than 32-bits.